wcfsetup/install/files/lib/system/background/job/TolerantCacheRebuildBackgroundJob.class.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace wcf\system\background\job;
+
+use wcf\system\cache\CacheHandler;
+use wcf\system\cache\tolerant\AbstractTolerantCache;
+
+/**
+ * @author Olaf Braun
+ * @copyright 2001-2025 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.2
+ */
+final class TolerantCacheRebuildBackgroundJob extends AbstractUniqueBackgroundJob
+{
+    public function __construct(
+        /** @var class-string<AbstractTolerantCache<array|object> */
+        public readonly string $cacheClass,
+        /** @var array<string, mixed> */
+        public readonly array $parameters = []
+    ) {
+    }
+
+    public function identifier(): string
+    {
+        $identifier = $this->cacheClass;
+        if (!empty($this->parameters)) {
+            $identifier .= '-' . CacheHandler::getInstance()->getCacheIndex($this->parameters);
+        }
+
+        return $identifier;
+    }
+
+    #[\Override]
+    public function newInstance(): static
+    {
+        return new TolerantCacheRebuildBackgroundJob($this->cacheClass, $this->parameters);
+    }
+
+    #[\Override]
+    public function queueAgain(): bool
+    {
+        return \class_exists($this->cacheClass);
+    }
+
+    #[\Override]
+    public function retryAfter()
+    {
+        return (new $this->cacheClass(...$this->parameters))->getLifetime();
```

Comment:
I don’t understand this decision, this value is meant to gracefully recover from transient failures by queueing it shortly afterwards. You’re basically removing this feature by setting the retry period to equal the maximum lifetime of the cache.

It’s arguably if the default value of 60 seconds is a proper choice here, maybe even something shorter like 30 seconds could work. But using the maximum lifetime doesn’t feel right here.